secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/JwsSigner.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.signer;
+
+import java.util.Map;
+
+import org.forgerock.util.promise.Promise;
+
+/**
+ * Abstraction of the signature methods to implement a code segment to sign messages.
+ * <p>
+ *     As result, the sign task produce a JWS Compact Serialization from JWS, a string composed of three parts encoded in Base64 Url Safe and separated by a dot ( . ), using
+ *     <a href='https://datatracker.ietf.org/doc/html/rfc7159'>JSON-based data structures</a>,
+ *     in accordance with: <a href='https://datatracker.ietf.org/doc/html/rfc7515'>RFC7515 JSON Web Signature (JWS)</a>
+ * </p>
```

Comment:
This is an implementation detail of the CompactSerializationJwsSigner not of the JwsSigner interface which could be used to create a Json Serialized Jws  (instead of a Compact one), see: https://datatracker.ietf.org/doc/html/rfc7515#section-7

Reply:
Moved to the implementation